context.go

```diff
@@ -118,6 +126,42 @@ func (c *Ctx) LoadProject(path string) (*Project, error) {
 	return p, nil
 }
 
+// resolveProjectRoot evaluates the root directory and does the following:
+//
+// If the passed path is a symlink outside GOPATH to a directory within a
+// GOPATH, the resolved full real path is returned.
+//
+// If the passed path is a symlink within a GOPATH, we return an error.
+//
+// If the passed path isn't a symlink at all, we just pass through.
+func (c *Ctx) resolveProjectRoot(path string) (string, error) {
+	// Determine if this path is a Symlink
+	l, err := os.Lstat(path)
+	if err != nil {
+		return "", errors.Wrap(err, "resolveProjectRoot")
+	}
+
+	// Pass through if not
+	if l.Mode()&os.ModeSymlink == 0 {
+		return path, nil
+	}
+
+	// Resolve path
+	resolved, err := filepath.EvalSymlinks(path)
+	if err != nil {
+		return "", errors.Wrap(err, "resolveProjectRoot")
+	}
+
+	// Determine if the symlink is within the GOPATH, in which case we're not
+	// sure how to resolve it.
+	if filepath.HasPrefix(path, c.GOPATH) {
```

Comment:
This search needs to be more expansive - it should check _all_ other GOPATHs, not just the selected one (see `newContext()` for how they're selected).

To minimize our exposure to other globals, let's update `newContext` to store all the GOPATHs (in addition to and separately from the selected one), then do the comparison here against what's stored locally on the struct.